x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/edit_policy.test.ts

```diff
@@ -672,66 +672,155 @@ describe('<EditPolicy />', () => {
-          ).toBe(true);
+
+        describe('v8 and above', () => {
+          beforeEach(async () => {
+            httpRequestsMockHelpers.setLoadPolicies([getDefaultHotPhasePolicy('my_policy')]);
+            httpRequestsMockHelpers.setListNodes({
+              nodesByAttributes: { test: ['123'] },
+              // On cloud, even if there are data_* roles set, the default, recommended allocation option should not
+              // be available.
+              nodesByRoles: { data_hot: ['123'] },
+              isUsingDeprecatedDataRoleConfig: true,
+            });
+            httpRequestsMockHelpers.setListSnapshotRepos({ repositories: ['found-snapshots'] });
+
+            await act(async () => {
+              testBed = await setup({
+                appServicesContext: {
+                  cloud: {
+                    isCloudEnabled: true,
+                  },
+                  license: licensingMock.createLicense({ license: { type: 'basic' } }),
+                  stackVersion: '8.0.0',
+                },
+              });
+            });
+
+            const { component } = testBed;
+            component.update();
+          });
+          test('hiding default, recommended option', async () => {
+            const { actions, find } = testBed;
+            await actions.warm.enable(true);
+            actions.warm.showDataAllocationOptions();
+
+            // Should show all options!!
+            expect(find('defaultDataAllocationOption').exists()).toBeTruthy();
+            expect(find('customDataAllocationOption').exists()).toBeTruthy();
+            expect(find('noneDataAllocationOption').exists()).toBeTruthy();
+          });
         });
       });
-      describe('existing policy', () => {
-        beforeEach(async () => {
-          httpRequestsMockHelpers.setLoadPolicies([getDefaultHotPhasePolicy('my_policy')]);
-          httpRequestsMockHelpers.setListNodes({
-            isUsingDeprecatedDataRoleConfig: false,
-            nodesByAttributes: { test: ['123'] },
-            nodesByRoles: { data: ['123'] },
+      describe('using data role config', () => {
+        describe('new policy', () => {
+          beforeEach(async () => {
+            // simulate creating a new policy
+            httpRequestsMockHelpers.setLoadPolicies([getDefaultHotPhasePolicy('')]);
+            httpRequestsMockHelpers.setListNodes({
+              isUsingDeprecatedDataRoleConfig: false,
+              nodesByAttributes: { test: ['123'] },
+              nodesByRoles: { data: ['123'] },
+            });
+            httpRequestsMockHelpers.setListSnapshotRepos({ repositories: ['found-snapshots'] });
+
+            await act(async () => {
+              testBed = await setup({ appServicesContext: { cloud: { isCloudEnabled: true } } });
+            });
+
+            const { component } = testBed;
+            component.update();
+          });
+          test('defaults searchable snapshot to true on cloud', async () => {
```

Comment:
This is just a new test, right? We're not testing behavior added in this PR?

Reply:
Right, the "cloud" tests were migrated over from the legacy "components" folder.